it/splunk/src/main/java/com/google/cloud/teleport/it/splunk/SplunkContainer.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.splunk;
+
+import java.time.Duration;
+import java.util.Map;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.images.builder.Transferable;
+import org.testcontainers.utility.DockerImageName;
+
+/** Constructs a Splunk container.
+ * <p>Tested on a Splunk version 8.2.
+ * <p>More information about docker-splunk can be found here:
+ * <p><a href=>"https://splunk.github.io/docker-splunk/"</a>
+ * */
+public class SplunkContainer extends GenericContainer<SplunkContainer> {
+  private static final Logger log = LoggerFactory.getLogger(SplunkContainer.class);
+
+  /** Splunk Default HTTP port. */
+  private static final int SPLUNK_INTERNAL_PORT = 8000;
+
+  /** Splunk Default HTTP Event Collector (HEC) port. */
+  private static final int SPLUNK_HEC_INTERNAL_PORT = 8088;
```

Comment:
Can we just use this on the RM? Ports are defined twice

Reply:
I did that because the goal is to eventually contribute to TestContainers which define ports in the container.

Reply:
Sounds good, thanks!